### ISSUE TITLE
Check in .maintenance.json

### ISSUE THREAD

**Issue by jonathanolson** [phetsims/perennial#134]
To allow switching between computers (after updateDependencies or before patches), and since only one of these should be used at a time, it would be great if .maintenance.json changed to a checked-in file. Noted in https://github.com/phetsims/perennial/issues/132 as:

> Should we be able to check in .maintenance.json? This could be helpful for improving collaboration or saving progress when the process takes a long time. It could also be problematic if there are multiple batch releases in progress concurrently.

The trickiest part of this would be whether to decide on auto-commits, or if we want to manually commit the file. (Presumably it would be in perennial?)

@mattpen and @zepumph, do you have thoughts/preferences on this?

**Comment by zepumph** [comment]
>after updateDependencies or before patches

Right, there is no reasonable way to have a cross computer solution before `updateDependencies`. I agree with your lifecycle timeline.

>The trickiest part of this would be whether to decide on auto-commits, or if we want to manually commit the file. (Presumably it would be in perennial?)

Yes that is a good question. I can see it either way. Probably the best long-term solution would be to have the .maintenance release patch checked in and committed automatically. Basically to have everything done for you without you needing to think about it.

If we now just have a single .maintenance.json per project, then I think we should add some logic also to mark that there is a current maintenance release in progress. I'm not entirely sure how that would work, but somehow disallowing the beginning commands to occur once you have already called `updateDependencies` to make sure that multiple maintenance releases don't get mixed up.

**Comment by mattpen** [comment]
I think it should be automated, if possible.

> make sure that multiple maintenance releases don't get mixed up.

Wouldn't calling Maintenance.reset() fix this?

**Comment by zepumph** [comment]
>Wouldn't calling Maintenance.reset() fix this?

Yes that makes sense. Forgive me since I don't understand this process as much as you two, but I would also want to see the following if there was already a maintenance release in progress:

```
Maintenance.startANewProcess()

> ERROR: sorry there is already a maintenance release going on, finish that one first. If this is a mistake then you can use `Maintenance.reset()` to start over.
```

**Comment by jonathanolson** [comment]
Because it needs to update the file after every simulation in many cases (so it can prevent errors), this would cause many hundreds of commits to perennial. Is that a major downside?

**Comment by zepumph** [comment]
@jonathanolson and I discussed this today while working on https://github.com/phetsims/chipper/issues/661. We thought one way to proceed could be to just check in the file after whole commands had finished. This has the drawback that if `updateDependencies` fails half way through, the maintenance file wouldn't be checked in, but mast on github would be partially co-opted by this maintenance release.

Basically the two main important things to me is to have a way to pass the process off to another person (in between commands like `updateDependencies` and `deployReleaseCandidates` and `deployProduction`), and also to have a way to lock out new maintenance releases once `updateDependencies` has been called.

**Comment by zepumph** [comment]
Unassigning for now, let me know if you need anything.
